superset/views/core.py

```diff
@@ -715,6 +715,68 @@ def shortner(self):
 appbuilder.add_view_no_menu(R)
 
 
+class SQLJsonParams:
```

Comment:
This class is a bit funky to me because it's super coupled with `request.form` which is super coupled with the that specific view. Conceptually, parsing `request.form` is typical view-related code.

Knowing that the goal here is to compartmentalize a chunk of code to make it more readable, maybe using an inline function (function defined in the view itself) makes most sense. Something named `_create_query_instance(request.form)` and maybe passing it `request.form` explicitly just for clarity, trying to make the function feel actually "functional".

Also to me assigning to self in the constructor and then to back to Query in `create_query` adds boilerplate and indirection. 

I think stuff dealing with the session in `save_query` also should live in view-land. Ideally the view itself has the main handle on the session / transaction, meaning ideally commits and rollbacks are in view scope.

Reply:
I agree, I don't like it either (the SQLJsonParams class), it was an idea around a data structure for all those `request.form` parameters, that could be passed around to other class methods